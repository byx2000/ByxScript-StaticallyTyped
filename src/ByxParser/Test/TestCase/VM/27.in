function int fib(int n)
{
	if (n == 1 || n == 2)
		return 1;
	return fib(n - 1) + fib(n - 2);
}

function int Gcd(int a, int b)
{
	int m = a;
	if (a > b)
	{
		m = b;
	}
	
	while (m != 0)
	{
		if (a % m == 0 && b % m == 0)
		{
			return m;
		}
		m = m - 1;
	}
	
	return 1;
}

function int Gcd2(int p, int q)
{
	if (q == 0) return p;
	int r = p % q;
	return Gcd2(q, r);
}

int g1;
int g2;
int g3;
int g4;
int g5;
int g6;
int g7;
int g8;
int g9;
int g10;
int g11;
int g12;
int g13;
int g14;
int g15;
int g16;

function void main()
{
	g1 = fib(1);
	g2 = fib(2);
	g3 = fib(3);
	g4 = fib(4);
	g5 = fib(10);
	g6 = fib(12);
	g7 = fib(17);
	g8 = fib(18);
	
	g9 = Gcd(12, 15);
	g10 = Gcd(17, 18);
	g11 = Gcd(223432, 914172);
	g12 = Gcd(7744, 5533);
	
	g13 = Gcd2(12, 15);
	g14 = Gcd2(17, 18);
	g15 = Gcd2(223432, 914172);
	g16 = Gcd2(7744, 5533);
}