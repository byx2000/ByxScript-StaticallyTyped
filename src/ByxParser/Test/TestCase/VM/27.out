i 1
i 1
i 2
i 3
i 55
i 144
i 1597
i 2584
i 3
i 1
i 4
i 11
i 3
i 1
i 4
i 11